kernel/src/platform/mod.rs

```diff
@@ -7,26 +7,90 @@ pub mod mpu;
 crate mod systick;
 
 /// Interface for individual boards.
+///
+/// Each board in their `main.rs` file should define a struct which implements
+/// this trait. This trait is the core for how syscall dispatching is handled.
```

Comment:
Maybe something more specific: "This trait is responsible for dispatching drivers for each system call number."